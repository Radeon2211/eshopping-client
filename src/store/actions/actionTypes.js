export const LOGIN_USER = 'LOGIN_USER';
export const LOGOUT_USER = 'LOGOUT_USER';
export const SET_PLACED_ORDERS = 'SET_PLACED_ORDERS';
export const SET_SELL_HISTORY = 'SET_SELL_HISTORY';
export const SET_PROFILE = 'SET_PROFILE';
export const UPDATE_TRANSACTION = 'UPDATE_TRANSACTION';
export const SET_PRODUCTS = 'SET_PRODUCTS';
export const SET_PRODUCT_DETAILS = 'SET_PRODUCT_DETAILS';
export const DELETE_PRODUCT_DETAILS = 'DELETE_PRODUCT_DETAILS';
export const DELETE_PRODUCT_FROM_LIST = 'DELETE_PRODUCT_FROM_LIST';
export const CLEAR_PRODUCTS = 'CLEAR_PRODUCTS';
export const FORM_START = 'FORM_START';
export const FORM_SUCCESS = 'FORM_SUCCESS';
export const FORM_FAIL = 'FORM_FAIL';
export const DATA_START = 'DATA_START';
export const DATA_SUCCESS = 'DATA_SUCCESS';
export const DATA_FAIL = 'DATA_FAIL';
export const SET_MESSAGE = 'SET_MESSAGE';
export const DELETE_MESSAGE = 'DELETE_MESSAGE';
export const SET_MODAL_OPEN_STATE = 'SET_MODAL_OPEN_STATE';
export const SET_MAX_QUANTITY_PER_PAGE = 'SET_MAX_QUANTITY_PER_PAGE';
