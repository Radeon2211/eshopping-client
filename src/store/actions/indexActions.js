export {
  registerUser,
  loginUser,
  fetchProfile,
  logoutUser,
  fetchOtherUser,
  setOtherUser,
  updateUser,
  changeDeliveryAddress,
  deleteAccount,
  addAdmin,
  removeAdmin,
  sendAccountVerificationLink,
  resetPassword,
} from './authActions';

export {
  fetchCart,
  addCartItem,
  updateCartItem,
  clearCart,
  removeCartItem,
  setTransaction,
  goToTransaction,
  buyProducts,
} from './tradeActions';

export { setModal, deleteMessage, changeProductsPerPage } from './uiActions';

export {
  addProduct,
  editProduct,
  fetchProducts,
  fetchProductDetails,
  setProductDetails,
  deleteProduct,
} from './productActions';

export { fetchOrders, fetchOrderDetails, setOrderDetails } from './orderActions';
