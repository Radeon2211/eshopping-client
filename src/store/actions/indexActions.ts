export {
  registerUser,
  loginUser,
  fetchProfile,
  logoutUser,
  fetchOtherUser,
  setOtherUser,
  updateUser,
  changeEmail,
  changeDeliveryAddress,
  deleteAccount,
  addAdmin,
  removeAdmin,
  sendAccountVerificationLink,
  resetPassword,
} from './authActions/authActions';

export {
  fetchCart,
  addCartItem,
  updateCartItem,
  clearCart,
  removeCartItem,
  setTransaction,
  goToTransaction,
  buyProducts,
} from './tradeActions/tradeActions';

export { setMessage, setModal, changeProductsPerPage } from './uiActions/uiActions';

export {
  addProduct,
  editProduct,
  fetchProducts,
  fetchProductDetails,
  setProductDetails,
  deleteProduct,
} from './productActions/productActions';

export { fetchOrders, fetchOrderDetails, setOrderDetails } from './orderActions/orderActions';
