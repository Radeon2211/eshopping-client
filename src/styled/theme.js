const theme = {
  colors: {
    light1: '#FAFAFA',
    light1Transparent: '#FAFAFA88',
    light2: '#DADFE1',
    light3: '#BFBFBF',
    light4: '#666',
    blue: '#1E8BC3',
    blueLight: '#22A7F0',
    blueDark: '#67809F',
    green: '#00A790',
    greenDark: '#009680',
    red: '#F22613',
  },
  shadows: {
    level1: '0 0 2px rgba(0, 0, 0, .1)',
    level2: '0 0 2px rgba(0, 0, 0, .3)',
    level3: '0 0 3px rgba(0, 0, 0, .3)',
    level4: '0 0 5px rgba(0, 0, 0, .3)',
    level5: '0 0 8px rgba(0, 0, 0, .3)',
  },
  fonts: {
    logo: 'Lobster, cursive',
    heading: 'Ubuntu, sans-serif',
    text: 'Lato, sans-serif',
  },
  spacings: {
    level1: '0.8rem',
    level2: '1.2rem',
    level3: '2.4rem',
    level4: '3.6rem',
    level5: '4.8rem',
  },
  zIndexes: {
    level1: 10,
    level2: 100,
    level3: 1000,
    level4: 10000,
  },
};

export default theme;
