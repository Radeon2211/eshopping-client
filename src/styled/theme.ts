const theme = {
  colors: {
    light1: '#FAFAFA',
    light1Transparent: '#FAFAFA88',
    light2: '#DADFE1',
    light3: '#BFBFBF',
    light4: '#777',
    blue: '#1E8BC3',
    blueLight: '#22A7F0',
    blueDark: '#67809F',
    green: '#06AD96',
    greenDark: '#009680',
    red: '#F22613',
    redLight: '#F94D2A',
    redDark: '#985A5A',
  },
  shadows: {
    level1: '0 0 3px rgba(0, 0, 0, 0.03)',
    level2: '0 0 3px rgba(0, 0, 0, 0.3)',
    level3: '0 0 5px rgba(0, 0, 0, 0.3)',
  },
  fonts: {
    logo: 'Lobster, cursive',
    heading: 'Noto Sans, sans-serif',
    text: 'Nunito, sans-serif',
  },
  spacings: {
    level1: '0.8rem',
    level2: '1.2rem',
    level3: '2.4rem',
    level4: '3.6rem',
    level5: '4.8rem',
  },
  zIndexes: {
    level1: 10,
    level2: 100,
    level3: 1000,
    level4: 10000,
    level5: 100000,
  },
  fontSizes: {
    level1: '1.3rem',
    level2: '1.4rem',
    level3: '1.5rem',
    level4: '1.7rem',
    level5: '2.2rem',
    level6: '2.6rem',
  },
  durations: {
    level1: 0.1,
    level2: 0.2,
    level3: 0.3,
  },
  lineHeights: {
    level1: '1',
    level2: '1.1',
    level3: '1.2',
    level4: '1.3',
    level5: '1.4',
    level6: '1.5',
  },
};

export type ThemeSpacing = keyof typeof theme.spacings;
export type ThemeZIndex = keyof typeof theme.zIndexes;
export type ThemeFontSize = keyof typeof theme.fontSizes;
export type ThemeDuration = keyof typeof theme.durations;
export type ThemeLineHeight = keyof typeof theme.lineHeights;

export default theme;
