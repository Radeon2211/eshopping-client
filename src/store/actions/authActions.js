import axios from '../../axios';
import * as actionTypes from './actionTypes';
import * as uiActions from './uiActions';
import { getErrorMessage } from '../../shared/utility';
import { DEFAULT_PATH } from '../../shared/constants';

export const setProfile = (profile) => ({
  type: actionTypes.SET_PROFILE,
  profile,
});

export const logout = () => ({
  type: actionTypes.LOGOUT_USER,
});

export const setDeliveryAddress = (deliveryAddress) => ({
  type: actionTypes.SET_DELIVERY_ADDRESS,
  deliveryAddress,
});

export const setOtherUser = (user) => ({
  type: actionTypes.SET_OTHER_USER,
  otherUser: user,
});

export const registerUser = (creds) => {
  return async (dispatch) => {
    dispatch(uiActions.formStart());
    const country = creds.country.value;
    const phone = `+${creds.phonePrefix.value} ${creds.phoneNumber}`;
    const contacts = [];
    if (!creds.hideEmail) contacts.push('email');
    if (!creds.hidePhone) contacts.push('phone');
    const correctCreds = {
      ...creds,
      country,
      phone,
      contacts,
    };
    delete correctCreds.hideEmail;
    delete correctCreds.hidePhone;
    delete correctCreds.phoneNumber;
    delete correctCreds.phonePrefix;
    try {
      const { data } = await axios.post('/users', { data: correctCreds });
      dispatch(setProfile(data.user));
      dispatch(uiActions.setAndDeleteMessage('Your account has been created successfully!'));
      dispatch(uiActions.formSuccess());
    } catch (error) {
      const errorMessage = getErrorMessage(error);
      dispatch(uiActions.formFail(errorMessage));
    }
  };
};

export const loginUser = (creds) => {
  return async (dispatch) => {
    dispatch(uiActions.formStart());
    try {
      const { data } = await axios.post('/users/login', { data: creds });
      dispatch(setProfile(data.user));
      dispatch(uiActions.writeChangeCartInfo(data.isDifferent));
      dispatch(uiActions.formSuccess());
    } catch (error) {
      const errorMessage = getErrorMessage(error);
      dispatch(uiActions.formFail(errorMessage));
    }
  };
};

export const fetchProfile = () => {
  return async (dispatch) => {
    try {
      const { data } = await axios.get('/users/me');
      dispatch(setProfile(data.user));
      dispatch(uiActions.writeChangeCartInfo(data.isDifferent));
    } catch (error) {
      dispatch(setProfile(null));
    }
  };
};

export const logoutUser = () => {
  return async (dispatch) => {
    try {
      await axios.post('/users/logout');
      dispatch(setProfile(null));
    } catch (error) {
      dispatch(uiActions.setAndDeleteMessage('Unable to logout. Something went wrong'));
    }
  };
};

export const updateUser = (creds, message) => {
  return async (dispatch) => {
    dispatch(uiActions.formStart());
    try {
      const { data } = await axios.patch('/users/me', creds);
      dispatch(setProfile(data.user));
      dispatch(uiActions.setAndDeleteMessage(message));
      dispatch(uiActions.formSuccess());
    } catch (error) {
      const errorMessage = getErrorMessage(error);
      dispatch(uiActions.formFail(errorMessage));
    }
  };
};

export const deleteAccount = (creds, history) => {
  return async (dispatch, getState) => {
    dispatch(uiActions.formStart());
    try {
      const { username } = getState().auth.profile;
      await axios.delete('/users/me', { data: creds });
      dispatch(setProfile(null));
      dispatch(
        uiActions.setAndDeleteMessage(`Your account has been deleted. Goodbye ${username}!`),
      );
      dispatch(uiActions.formSuccess());
      history.replace(DEFAULT_PATH);
    } catch (error) {
      const errorMessage = getErrorMessage(error);
      dispatch(uiActions.formFail(errorMessage));
    }
  };
};

export const changeDeliveryAddress = (creds) => {
  return async (dispatch) => {
    dispatch(uiActions.formStart());
    try {
      const { onlyTheseOrders } = creds;
      // eslint-disable-next-line no-param-reassign
      delete creds.onlyTheseOrders;

      if (onlyTheseOrders) {
        dispatch(setDeliveryAddress(creds));
      } else {
        const { data } = await axios.patch('/users/me', creds);
        dispatch(setProfile(data.user));
        dispatch(uiActions.setAndDeleteMessage('Delivery address has been saved in your profile'));
      }
      dispatch(uiActions.formSuccess());
    } catch (error) {
      const errorMessage = getErrorMessage(error);
      dispatch(uiActions.formFail(errorMessage));
    }
  };
};

export const fetchOtherUser = (username) => {
  return async (dispatch) => {
    dispatch(uiActions.dataStart());
    try {
      const { data } = await axios.get(`/users/${username}`);
      dispatch(setOtherUser(data.profile));
      dispatch(uiActions.dataEnd());
    } catch (error) {
      const errorMessage = getErrorMessage(error);
      dispatch(uiActions.setAndDeleteMessage(errorMessage));
      dispatch(setOtherUser(null));
      dispatch(uiActions.dataEnd());
    }
  };
};
