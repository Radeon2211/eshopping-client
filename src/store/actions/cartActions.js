import axios from '../../axios';
import * as actionTypes from './actionTypes';
import * as uiActions from './uiActions';
import { getErrorMessage } from '../../shared/utility';
import { modalTypes } from '../../shared/constants';

let updateCartItemReqCounter = 0;

export const setCart = (cart) => ({
  type: actionTypes.SET_CART,
  cart,
});

export const fetchCart = () => {
  return async (dispatch) => {
    dispatch(uiActions.cartStart());
    try {
      const { data } = await axios.get('/cart');
      dispatch(uiActions.cartEnd());
      dispatch(setCart(data.cart));
      dispatch(uiActions.writeChangeCartInfo(data.isDifferent));
    } catch (error) {
      const errorMessage = getErrorMessage(error);
      dispatch(uiActions.setAndDeleteMessage(errorMessage));
      dispatch(uiActions.cartEnd());
      dispatch(setCart(null));
    }
  };
};

export const addCartItem = (item) => {
  return async (dispatch) => {
    dispatch(uiActions.cartStart());
    try {
      const { data } = await axios.patch('/cart/add', item);
      dispatch(uiActions.cartEnd());
      dispatch(setCart(data.cart));
      dispatch(uiActions.setModal(true, modalTypes.CART_ITEM_ADDED));
      dispatch(uiActions.writeChangeCartInfo(data.isDifferent));
    } catch (error) {
      const errorMessage = getErrorMessage(error);
      dispatch(uiActions.setAndDeleteMessage(errorMessage));
      dispatch(uiActions.cartEnd());
    }
  };
};

export const updateCartItem = (itemId, action, quantity) => {
  return async (dispatch) => {
    dispatch(uiActions.cartStart());
    try {
      updateCartItemReqCounter += 1;
      const quantityParam = quantity ? `&quantity=${quantity}` : '';
      const { data } = await axios.patch(`/cart/${itemId}/update?action=${action}${quantityParam}`);
      updateCartItemReqCounter -= 1;
      if (updateCartItemReqCounter <= 0) {
        dispatch(uiActions.cartEnd());
        dispatch(setCart(data.cart));
        dispatch(uiActions.writeChangeCartInfo(data.isDifferent));
      }
    } catch (error) {
      const errorMessage = getErrorMessage(error);
      dispatch(uiActions.setAndDeleteMessage(errorMessage));
      dispatch(uiActions.cartEnd());
    }
  };
};

export const clearCart = () => {
  return async (dispatch) => {
    dispatch(uiActions.cartStart());
    try {
      await axios.patch('/cart/clear');
      dispatch(uiActions.cartEnd());
      dispatch(setCart([]));
    } catch (error) {
      const errorMessage = getErrorMessage(error);
      dispatch(uiActions.setAndDeleteMessage(errorMessage));
      dispatch(uiActions.cartEnd());
    }
  };
};

export const removeCartItem = (itemId) => {
  return async (dispatch) => {
    dispatch(uiActions.cartStart());
    try {
      const { data } = await axios.patch(`/cart/${itemId}/remove`);
      dispatch(uiActions.cartEnd());
      dispatch(setCart(data.cart));
      dispatch(uiActions.writeChangeCartInfo(data.isDifferent));
    } catch (error) {
      const errorMessage = getErrorMessage(error);
      dispatch(uiActions.setAndDeleteMessage(errorMessage));
      dispatch(uiActions.cartEnd());
    }
  };
};
