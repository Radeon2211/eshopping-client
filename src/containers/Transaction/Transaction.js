import React, { useEffect, useCallback } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';
import * as actions from '../../store/actions/indexActions';
import SideBySide from '../../components/UI/SideBySide';
import FlexWrapper from '../../components/UI/FlexWrapper';
import PlainPanel from '../../components/UI/Panels/PlainPanel';
import StickyPanel from '../../components/UI/Panels/StickyPanel';
import DeliveryAddressSection from './DeliveryAddressSection/DeliveryAddressSection';
import Heading from '../../components/UI/Heading/Heading';
import ToPayInfo from '../../components/UI/ToPayInfo/ToPayInfo';
import CartAndTransactionItems from '../../components/CartAndTransactionItems/CartAndTransactionItems';
import Button from '../../components/UI/Button/Button';
import { itemTypes, modalTypes } from '../../shared/constants';
import { roundOverallPrice, scrollToTop } from '../../shared/utility/utility';
import MetaDescriptor from '../../components/MetaDescriptor/MetaDescriptor';

const Transaction = () => {
  const history = useHistory();

  const transaction = useSelector((state) => state.auth.transaction);

  const dispatch = useDispatch();
  const onSetModal = useCallback((modalContent) => dispatch(actions.setModal(modalContent)), [
    dispatch,
  ]);
  const onSetTransaction = useCallback((items) => dispatch(actions.setTransaction(items)), [
    dispatch,
  ]);

  useEffect(() => {
    scrollToTop();
    if (!transaction || transaction?.length <= 0) {
      history.replace('/cart');
    }
    return () => onSetTransaction(undefined);
  }, [history, transaction, onSetTransaction]);

  let content = null;
  if (transaction?.length > 0) {
    const toPayValue = transaction.reduce((acc, { price, quantity }) => {
      return acc + price * quantity;
    }, 0);
    const roundedToPayValue = roundOverallPrice(toPayValue);

    content = (
      <>
        <Heading variant="h3">Transaction</Heading>
        <SideBySide proportion="3/1" makeVerticalWhen={1200}>
          <FlexWrapper direction="column" align="stretch" spacing="3">
            <DeliveryAddressSection onSetModal={onSetModal} />
            <PlainPanel>
              <Heading variant="h4">Products</Heading>
              <CartAndTransactionItems items={transaction} type={itemTypes.TRANSACTION} />
            </PlainPanel>
          </FlexWrapper>
          <StickyPanel>
            <FlexWrapper direction="column" spacing="2">
              <ToPayInfo value={roundedToPayValue} />
              <Button filled clicked={() => onSetModal(modalTypes.BUY_PRODUCTS)}>
                I buy and pay
              </Button>
            </FlexWrapper>
          </StickyPanel>
        </SideBySide>
      </>
    );
  }

  return (
    <>
      <MetaDescriptor title="Transaction summary - E-Shopping" />
      {content}
    </>
  );
};

export default Transaction;
