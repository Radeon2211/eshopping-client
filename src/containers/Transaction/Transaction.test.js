import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Transaction from './Transaction';
import theme from '../../styled/theme';
import {
  createTransactionAndOrderProdItem,
  defaultDeliveryAddress,
} from '../../shared/testUtility/testUtility';
import * as actions from '../../store/actions/indexActions';

const mockStore = configureMockStore([thunk]);

const setUp = (transaction) => {
  const store = mockStore({
    auth: {
      transaction,
      deliveryAddress: defaultDeliveryAddress,
    },
  });
  store.dispatch = jest.fn();

  const history = {
    listen: jest.fn(),
    createHref: jest.fn(),
    location: { pathname: '/transaction' },
    replace: jest.fn(),
  };

  return {
    ...render(
      <Router history={history}>
        <Provider store={store}>
          <ThemeProvider theme={theme}>
            <Transaction />
          </ThemeProvider>
        </Provider>
      </Router>,
    ),
    store,
    history,
  };
};

afterEach(cleanup);

describe('<Transaction />', () => {
  describe('check how renders', () => {
    it('should render everything correctly with two items from one user', () => {
      const transaction = [
        createTransactionAndOrderProdItem({
          productId: 'p1',
          sellerUsername: 'user1',
          price: 10.6,
          quantity: 4,
          name: 'product1',
          photo: false,
        }),
        createTransactionAndOrderProdItem({
          productId: 'p2',
          sellerUsername: 'user1',
          price: 299.98,
          quantity: 6,
          name: 'product2',
          photo: false,
        }),
      ];
      const { asFragment } = setUp(transaction);
      expect(asFragment()).toMatchSnapshot();
    });

    it('should render nothing if transaction is empty', () => {
      const { asFragment } = setUp([]);
      expect(asFragment()).toMatchSnapshot();
    });

    it('should render nothing if transaction is falsy', () => {
      const { asFragment } = setUp(undefined);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('check useEffect()', () => {
    it('should call replace if transaction is falsy', () => {
      const { history } = setUp(undefined);
      expect(history.replace).toHaveBeenCalledWith('/cart');
    });

    it('should call replace if transaction length is 0', () => {
      const { history } = setUp([]);
      expect(history.replace).toHaveBeenCalledWith('/cart');
    });

    it('should call setTransaction() when unmounting', () => {
      const { store, unmount } = setUp();
      expect(store.dispatch).not.toHaveBeenCalled();
      unmount();
      expect(store.dispatch).toHaveBeenCalledWith(actions.setTransaction(undefined));
    });
  });
});
