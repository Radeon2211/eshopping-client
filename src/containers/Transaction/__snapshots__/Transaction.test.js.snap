// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Transaction /> check how renders should render everything correctly with two items from one user 1`] = `
<DocumentFragment>
  <h3
    class="sc-eCssSg ceYvmz"
  >
    Transaction
  </h3>
  <div
    class="sc-bdfBwQ gbtjgb"
  >
    <div
      class="sc-gsTCUz bwlxQF"
      direction="column"
      spacing="3"
      width="100%"
      wrap="nowrap"
    >
      <section
        class="sc-dlfnbm ctISJv"
        data-testid="DeliveryAddressSection"
      >
        <div
          class="sc-gsTCUz evuYyy"
          direction="column"
          spacing="3"
          width="100%"
          wrap="nowrap"
        >
          <h4
            class="sc-eCssSg gLASfs"
          >
            Delivery address
          </h4>
          <div
            class="sc-gsTCUz iVaLBf"
            direction="column"
            spacing="1"
            width="100%"
            wrap="nowrap"
          >
            <span
              class="sc-fubCfw osNBQ"
              color=""
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="0px"
              transform=""
            >
              firstName lastName
            </span>
            <span
              class="sc-fubCfw osNBQ"
              color=""
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="0px"
              transform=""
            >
              street
            </span>
            <span
              class="sc-fubCfw osNBQ"
              color=""
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="0px"
              transform=""
            >
              zipCode city
            </span>
            <span
              class="sc-fubCfw osNBQ"
              color=""
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="0px"
              transform=""
            >
              country
            </span>
            <span
              class="sc-fubCfw osNBQ"
              color=""
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="0px"
              transform=""
            >
              phone
            </span>
          </div>
          <button
            class="sc-jSgupP kPCWJS"
            color="blue"
            type="button"
          >
            Change address
          </button>
        </div>
      </section>
      <section
        class="sc-dlfnbm ctISJv"
      >
        <h4
          class="sc-eCssSg gLASfs"
        >
          Products
        </h4>
        <div
          class="sc-bkzZxe hkHVNa"
        >
          <span
            class="sc-fubCfw jalAIu"
            color=""
            display="block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            <span>
              seller 
            </span>
            <a
              data-testid="CartAndTransactionItems-seller-link"
            >
              <span
                class="sc-fodVxV fCxCGe"
              >
                user1
              </span>
            </a>
          </span>
          <div
            class="sc-gsTCUz dMhtIX sc-jrAGrp jjgxPm"
            data-testid="TransactionAndOrderProdItem"
            direction="row"
            spacing="2"
            width="100%"
            wrap="nowrap"
          >
            <a
              data-testid="TransactionAndOrderProdItem-product-link-photo"
            >
              <div
                class="sc-gsTCUz cTtigW sc-crrsfI eexKSB"
                data-testid="ProductThumbnail"
                direction="row"
                height="5"
                spacing=""
                width="5"
                wrap="nowrap"
              >
                <img
                  alt="product1"
                  data-testid="ProductThumbnail-img"
                  src="no-photo.png"
                />
              </div>
            </a>
            <div
              class="sc-kEjbxe gFzfnQ"
            >
              <span
                class="name"
              >
                <a
                  data-testid="TransactionAndOrderProdItem-product-link-name"
                >
                  product1
                </a>
              </span>
              <div
                class="sc-gsTCUz erbZJc sc-iqHYGH bCSkmx"
                direction="row"
                spacing=""
                width="100%"
                wrap="nowrap"
              >
                <span
                  class="sc-fubCfw eLnxgP"
                  color="#777"
                  display="inline-block"
                  overflow="visible"
                  size="2"
                  spacing="0px"
                  transform=""
                >
                  4 x $10.60
                </span>
                <span
                  class="sc-fubCfw bvkSGA"
                  color=""
                  display="inline-block"
                  overflow="visible"
                  size="4"
                  spacing="0px"
                  transform=""
                >
                  $42.40
                </span>
              </div>
            </div>
          </div>
          <div
            class="sc-gsTCUz dMhtIX sc-jrAGrp jjgxPm"
            data-testid="TransactionAndOrderProdItem"
            direction="row"
            spacing="2"
            width="100%"
            wrap="nowrap"
          >
            <a
              data-testid="TransactionAndOrderProdItem-product-link-photo"
            >
              <div
                class="sc-gsTCUz cTtigW sc-crrsfI eexKSB"
                data-testid="ProductThumbnail"
                direction="row"
                height="5"
                spacing=""
                width="5"
                wrap="nowrap"
              >
                <img
                  alt="product2"
                  data-testid="ProductThumbnail-img"
                  src="no-photo.png"
                />
              </div>
            </a>
            <div
              class="sc-kEjbxe gFzfnQ"
            >
              <span
                class="name"
              >
                <a
                  data-testid="TransactionAndOrderProdItem-product-link-name"
                >
                  product2
                </a>
              </span>
              <div
                class="sc-gsTCUz erbZJc sc-iqHYGH bCSkmx"
                direction="row"
                spacing=""
                width="100%"
                wrap="nowrap"
              >
                <span
                  class="sc-fubCfw eLnxgP"
                  color="#777"
                  display="inline-block"
                  overflow="visible"
                  size="2"
                  spacing="0px"
                  transform=""
                >
                  6 x $299.98
                </span>
                <span
                  class="sc-fubCfw bvkSGA"
                  color=""
                  display="inline-block"
                  overflow="visible"
                  size="4"
                  spacing="0px"
                  transform=""
                >
                  $1,799.88
                </span>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
    <section
      class="sc-hKgILt ciIrWM"
    >
      <section
        class="sc-dlfnbm ctISJv"
      >
        <div
          class="sc-gsTCUz ewxwoI"
          direction="column"
          spacing="2"
          width="100%"
          wrap="nowrap"
        >
          <div
            class="sc-gsTCUz dYcWSu"
            direction="row"
            spacing="1"
            width="100%"
            wrap="wrap"
          >
            <span
              class="sc-fubCfw bvkSGA"
              color=""
              display="inline-block"
              overflow="visible"
              size="4"
              spacing="0px"
              transform=""
            >
              To pay
            </span>
            <span
              class="sc-fubCfw cQoALv"
              color=""
              display="inline-block"
              overflow="visible"
              size="6"
              spacing="1px"
              transform=""
            >
              $1,842.28
            </span>
          </div>
          <button
            class="sc-jSgupP jrAVxS"
            color="blue"
            type="button"
          >
            I buy and pay
          </button>
        </div>
      </section>
    </section>
  </div>
</DocumentFragment>
`;

exports[`<Transaction /> check how renders should render nothing if transaction is empty 1`] = `<DocumentFragment />`;

exports[`<Transaction /> check how renders should render nothing if transaction is falsy 1`] = `<DocumentFragment />`;
